wcfsetup/install/files/lib/system/devtools/package/DevtoolsPackageArchive.class.php

```diff
@@ -39,13 +37,55 @@ public function openArchive()
         $projectDir = FileUtil::addTrailingSlash(
             FileUtil::unifyDirSeparator(\realpath(\dirname($this->packageXmlPath)))
         );
-        $readFiles = DirectoryUtil::getInstance($projectDir)->getFiles(
-            \SORT_ASC,
-            // ignore folders whose contents are delivered as archives by default
-            // and ignore dotfiles and dotdirectories
-            Regex::compile('^' . \preg_quote($projectDir) . '(acptemplates|files|templates|\.)'),
-            true
+
+        $it = new \RecursiveIteratorIterator(
+            new \RecursiveCallbackFilterIterator(
+                new \RecursiveDirectoryIterator($projectDir),
+                static function (\SplFileInfo $current, string $key, \RecursiveDirectoryIterator $it) {
+                    // Skip '.' and '..'.
+                    if ($it->isDot()) {
+                        return false;
+                    }
+
+                    // Skip hidden files.
+                    if ($current->getFilename()[0] === '.') {
+                        return false;
+                    }
+
+                    if ($current->isDir()) {
+                        // Check if we are in the project root.
+                        if ($it->getSubPath() === '') {
+                            // Skip acptemplates / files / templates.
+                            if (
+                                \preg_match(
+                                    '/^(acptemplates|files|templates)(_[a-z0-9]+)?$/',
```

Comment:
This change narrows down the previous regular expression. I assume that you checked that the second group matches the rules for abbreviations used elsewhere.

Reply:
The abbreviations are derived from the package name (a-z0-9_-), turned into PHP constants (not allowing for the hyphen), thus the only thing that would be missing is the underscore which I expect to be reasonably not used :smiley: